src/backend/distributed/sql/udfs/fix_partition_shard_index_names/latest.sql

```diff
@@ -0,0 +1,28 @@
+CREATE FUNCTION pg_catalog.fix_partition_shard_index_names(table_name regclass)
+  RETURNS void
+  LANGUAGE C STRICT
+  AS 'MODULE_PATHNAME', $$fix_partition_shard_index_names$$;
+COMMENT ON FUNCTION pg_catalog.fix_partition_shard_index_names(table_name regclass)
+  IS 'fix index names on partition shards';
+
+CREATE OR REPLACE FUNCTION pg_catalog.fix_partition_shard_index_names()
```

Comment:
Makes sense, I will rename it.

I think we use CREATE OR REPLACE to help with downgrading/upgrading. This is the first time we introduce this function but in case we make some change in the future, we don't need to do DROP existing, CREATE new, we just do CREATE OR REPLACE. Same goes for downgrading: we don't need to do DROP new, CREATE old.